src/export/exporter.ts

```diff
@@ -42,16 +56,36 @@ export abstract class AzureMonitorBaseExporter implements BaseExporter {
       };
 
       if (err) {
+        // Request failed -- always retry
         this._logger.error(err.message);
         this._persister.push(envelopes, persistCb);
-      } else if (isRetriable(statusCode)) {
+      } else if (statusCode === 200) {
+        // Success -- @todo: start retry timer
+        if (!this._retryTimer) {
+          this._retryTimer = setTimeout(() => {
+            this._retryTimer = null;
+            this._sendFirstPersistedFile();
+          }, this._options.batchSendRetryIntervalMs);
+          this._retryTimer.unref();
+        }
+        resultCallback(ExportResult.SUCCESS);
+      } else if (statusCode && isRetriable(statusCode)) {
+        // Failed -- persist failed data
```

Comment:
Do you need to return FAILED_RETRYABLE here?

Reply:
It is currently called inside called inside `persistCb`. Will add a comment to clarify here.